File: news/src/main/java/com/leory/badminton/news/mvp/ui/fragment/PlayerMatchFragment.java
package com.leory.badminton.news.mvp.ui.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.leory.badminton.news.R;
import com.leory.badminton.news.R2;
import com.leory.badminton.news.di.component.PlayerDetailComponent;
import com.leory.badminton.news.mvp.contract.PlayerContract;
import com.leory.badminton.news.mvp.model.bean.PlayerMatchBean;
import com.leory.badminton.news.mvp.presenter.PlayerMatchPresenter;
import com.leory.badminton.news.mvp.ui.adapter.PlayerMatchAdapter;
import com.leory.badminton.news.mvp.ui.widget.decoration.MatchDateItemDecoration;
import com.leory.commonlib.base.BaseLazyLoadFragment;
import com.leory.commonlib.base.delegate.IComponent;
import com.leory.commonlib.di.scope.FragmentScope;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * Describe : 运动员比赛结果
 * Author : leory
 * Date : 2019-06-11
 */
@FragmentScope
public class PlayerMatchFragment extends BaseLazyLoadFragment<PlayerMatchPresenter> implements PlayerContract.MatchView {

    PlayerMatchAdapter adapter;
    @BindView(R2.id.rcv)
    RecyclerView rcv;
    @BindView(R2.id.progress)
    FrameLayout progress;

    public static PlayerMatchFragment newInstance() {
        return new PlayerMatchFragment();
    }

    @Override
    public IComponent setupActivityComponent(IComponent component) {
        ((PlayerDetailComponent) component).buildPlayerMatchComponent()
                .view(this)
                .build()
                .inject(this);
        return super.setupActivityComponent(component);
    }

    @Override
    protected void lazyLoadData() {
        presenter.requestData(null);
    }

    @Override
    public View initView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_player_match, container, false);
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        rcv.setLayoutManager(new LinearLayoutManager(getContext()));
        rcv.addItemDecoration(new MatchDateItemDecoration(getContext()));
        rcv.setAdapter(adapter = new PlayerMatchAdapter(new ArrayList<>()));
    }

    @Override
    public void showLoading() {
        progress.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideLoading() {
        progress.setVisibility(View.GONE);
    }

    @Override
    public void showMessage(@NonNull String message) {

    }

    @Override
    public void showMatchData(List<PlayerMatchBean> data) {
        adapter.setNewData(data);
    }

}

File: news/src/main/java/com/leory/badminton/news/mvp/ui/fragment/PlayerMatchFragment.kt
package com.leory.badminton.news.mvp.ui.fragment

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.leory.badminton.news.R
import com.leory.badminton.news.di.component.PlayerDetailComponent
import com.leory.badminton.news.mvp.contract.PlayerContract
import com.leory.badminton.news.mvp.model.bean.PlayerMatchBean
import com.leory.badminton.news.mvp.presenter.PlayerMatchPresenter
import com.leory.badminton.news.mvp.ui.adapter.PlayerMatchAdapter
import com.leory.badminton.news.mvp.ui.widget.decoration.MatchDateItemDecoration
import com.leory.badminton.news.mvp.ui.widget.spinner.SpinnerPopView
import com.leory.commonlib.base.BaseLazyLoadFragment
import com.leory.commonlib.base.delegate.IComponent
import com.leory.commonlib.di.scope.FragmentScope
import kotlinx.android.synthetic.main.fragment_player_match.*
import java.util.*

/**
 * Describe : 运动员比赛结果
 * Author : leory
 * Date : 2019-06-11
 */
@FragmentScope
class PlayerMatchFragment : BaseLazyLoadFragment<PlayerMatchPresenter>(), PlayerContract.MatchView {
    companion object {
        @JvmStatic
        fun newInstance(): PlayerMatchFragment {
            return PlayerMatchFragment()
        }
    }

    val adapter: PlayerMatchAdapter by lazy {
        PlayerMatchAdapter(ArrayList())
    }

    override fun setupActivityComponent(component: IComponent): IComponent {
        (component as PlayerDetailComponent).buildPlayerMatchComponent()
                .view(this)
                .build()
                .inject(this)
        return super.setupActivityComponent(component)
    }

    override fun lazyLoadData() {
        presenter?.requestData(null)
    }

    override fun initView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_player_match, container, false)
    }

    override fun initData(savedInstanceState: Bundle?) {
        rcv.layoutManager = LinearLayoutManager(context)
        rcv.addItemDecoration(MatchDateItemDecoration(context!!))
        rcv.adapter = adapter
        val years = arrayOf("2020", "2019", "2018")
        spinner_year.initData(listOf(*years), 0)
        spinner_year.setOnSelectListener(object : SpinnerPopView.OnSelectListener {
            override fun onItemClick(pos: Int, name: String) {
                presenter?.requestData(spinner_year.selectName)
            }
        })
        val types = arrayOf("全部", "精简")
        spinner_type.initData(listOf(*types), 0)
        spinner_type.setOnSelectListener(object : SpinnerPopView.OnSelectListener {
            override fun onItemClick(pos: Int, name: String) {
                presenter?.setIsShort(pos == 1)
            }
        })
    }

    override fun showLoading() {
        progress.visibility = View.VISIBLE
    }

    override fun hideLoading() {
        progress.visibility = View.GONE
    }

    override fun showMessage(message: String) {

    }

    override fun showMatchData(data: List<PlayerMatchBean>) {
        adapter.setNewData(data)
    }


}
